### ISSUE TITLE
Enhancement: Schema Evolution - currently change in schema causes the runtime to crash

### ISSUE THREAD

**Issue by digadeesh** [spiceai/spiceai#1480]
**Describe the bug**
Change in schema in file based acceleration table causes error (maybe by design?)

**To Reproduce**

load a different dataset into a file based accelerator with the same name.  The following error is logged:

2024-05-21T23:59:46.150442Z INFO runtime::opentelemetry: Spice Runtime OpenTelemetry listening on 127.0.0.1:50052
2024-05-21T23:59:46.150760Z INFO runtime::accelerated_table::refresh: Loading data for dataset taxi_trips
2024-05-21T23:59:46.789811Z ERROR runtime::accelerated_table::refresh: Error adding data for my_table: External error: Unable to insert into duckdb table: Binder Error: table my_table has 6 columns but 16 values were supplied

**Expected behavior**
Table is dropped and recreated in the DB with the new data.

**Screenshots**

**Additional context**
Potentially we don't want to do this.  But schema changes from source can happen and it seems we don't handle this well overall.

**Comment by phillipleblanc** [comment]
I don't think we should drop data in an accelerator by default - it could mean data loss/time loss for a misconfiguration. I.e. if I have a large file that I've accelerated that took several hours to populate, and then I accidentally misconfigure my spicepod to point to this file from a different dataset, then it would wipe out that data.

**Comment by phillipleblanc** [comment]
Goal: What do we want to handle for schema evolution

**Comment by phillipleblanc** [comment]
I can see 3 different things that mean "schema evolution":

1. A persistent accelerator (file-based or Postgres) has a different schema than the federated table it is connected to at startup. (This is what this issue is describing)
2. An accelerator, upon refresh or CDC, discovers that the schema of the data it received is different from the schema it started up with.
3. A federated table's schema becomes outdated because the source changes its schema. Spice would return the correct data in the new schema, but the schema registered in DataFusion would be outdated.

I think the current behavior of returning an error for 1) is correct - we don't know if the schema actually changed or there was a misconfiguration. The user needs to perform a manual action to remove the DB file (or drop the table) to allow Spice to startup. Alternatively, they could issue the correct `ALTER TABLE..` statements to make the schemas match. Neither of these steps we want Spice to do automatically, since we don't know the intent of the user.

For 2) we can be more confident this was an intention change made by the operator that we need to respond to. The optimal course would be for Spice to perform a schema comparison and issue the correct `ALTER TABLE..` statements to make the schemas match. This could be done for relatively simple operations like column addition, column removal, etc. Detecting column renames or other more advanced operations would be harder.
So I still think there are two possible options when we detect 2:
- 2a: Try to automatically apply the correct `ALTER TABLE..` statements to make the schemas match. If we can't, then return an error and require user intervention.
- 2b: Drop the existing table and recreate it with the new schema. Will always work, but has the drawback of needing to rebuild the accelerator table again.

3) We could do periodic schema checks on the federated table and update the DF schema when Spice detects changes. Not hard to do, but also has some design considerations. This would behave similarly to Dremio's metadata refresh, and we might want to make the schema inference interval configurable (or disabled if they don't care).
